src/raftstore/store/read_queue.rs

```diff
@@ -70,9 +70,35 @@ pub struct ReadIndexQueue {
     handled_cnt: usize,
     // map[uuid] -> offset in `reads`.
     contexts: HashMap<Uuid, usize>,
+
+    retry_countdown: usize,
 }
 
 impl ReadIndexQueue {
+    /// Check it's necessary to retry pending read requests or not.
+    /// Return true if all such conditions are satisfied:
+    /// 1. an election timeout elapsed from the last request push;
+    /// 2. an election timeout elapsed from the last retry;
+    /// 3. there are still unresolved requests in the queue.
+    pub fn check_needs_retry(&mut self, cfg: &Config) -> bool {
+        if self.reads.len() == self.ready_cnt {
+            return false;
+        }
+
+        if self.retry_countdown == usize::MAX {
+            self.retry_countdown = cfg.raft_election_timeout_ticks.checked_sub(1).unwrap();
```

Comment:
Election timeout tick has to be greater than 0. count down can be zero after minusing one.